apollo-mockserver/src/commonTest/kotlin/com/apollographql/apollo3/mockserver/test/MockServerCommonTest.kt

```diff
@@ -1,12 +1,36 @@
 package com.apollographql.apollo3.mockserver.test
 
+import com.apollographql.apollo3.annotations.ApolloExperimental
+import com.apollographql.apollo3.api.http.HttpMethod
+import com.apollographql.apollo3.api.http.HttpRequest
+import com.apollographql.apollo3.api.http.HttpResponse
+import com.apollographql.apollo3.exception.ApolloNetworkException
+import com.apollographql.apollo3.mockserver.MockDispatcher
+import com.apollographql.apollo3.mockserver.MockRecordedRequest
+import com.apollographql.apollo3.mockserver.MockResponse
+import com.apollographql.apollo3.mockserver.MockServer
 import com.apollographql.apollo3.mockserver.readRequest
+import com.apollographql.apollo3.network.http.DefaultHttpEngine
+import com.apollographql.apollo3.testing.runTest
 import okio.Buffer
 import kotlin.test.Test
 import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
 import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
 
+@OptIn(ApolloExperimental::class)
 class MockServerCommonTest {
+  private lateinit var mockServer: MockServer
+
+  private fun setUp() {
+    mockServer = MockServer()
+  }
+
+  private suspend fun tearDown() {
+    mockServer.stop()
+  }
```

Comment:
Can we split this file in 2? 

* `ReadRequestTest` (doesn't require a mockServer)
* `MockServerTest` (requires a mockServer)

Reply:
3 actually 😇 ?

* `ReadRequestTest` (doesn't require a mockServer)
* `EnqueueTest` (requires a default mockServer) 
* `CustomDispatcherTest` (requires a custom mockServer)